tests/unit/utils/test_misc.py

```diff
@@ -40,13 +41,16 @@ def test_find_application_ok_c(self):
     def test_find_application_ok_rust(self):
         device, sdk, appname = "device", "rust", "rustapp"
         with temporary_directory() as dir_path:
-            tmp_dir = (dir_path / "target" / device / "release")
+            os.chdir(dir_path)
```

Comment:
Instead of doing `os.chdir(<path>); subprocess.check_output(cmd)`, you can just do `subprocess.check_output(cmd, cwd=<path>)`.